spec/classes/config_spec.rb

```diff
@@ -1,5 +1,140 @@
+            end
+
+            it 'configures systemd' do
+              should contain_file('/etc/sysconfig/marathon')
+                         .with_content(/^JAVA_OPTS=-Xmx512m -Dlogback.configurationFile=file:\/etc\/marathon\/logback.xml$/)
+                         .without_content(/JAVA_HOME/)
+
+              should contain_file('/etc/systemd/system/marathon.service')
+                         .with_ensure('absent')
+              should contain_exec('systemctl-daemon-reload_marathon')
+            end
+
+          else
+            it 'is an unsupported OS' do
+              fail("#{facts[:osfamily]} is unsupported")
+            end
+        end
+      end
+
+      context 'with lots of custom params' do
+
+        let(:params) { {
+            :master => 'zk://foo:2181/mesos',
+            :zookeeper => 'zk://foo:2181/marathon',
+            :options => {
+                'some_options' => 'with-value',
+                'some_other_options' => 'with-other-value',
+            },
+            :env_var => {
+                'foo' => 'bar',
+                'f00' => 'b4r',
+            },
+            :java_home => '/opt/some/java/home',
+            :java_opts => '-Xmx1024m',
+            :ulimit => 9001,
+        } }
+
+        it { should compile }
+
+        it 'stores some config in /etc/marathon/conf' do
+          should contain_mesos__property('marathon_master')
+                     .with_value('zk://foo:2181/mesos')
+          should contain_mesos__property('marathon_zk')
+                     .with_value('zk://foo:2181/marathon')
+          should contain_mesos__property('marathon_some_options')
+                     .with_value('with-value')
+          should contain_mesos__property('marathon_some_other_options')
+                     .with_value('with-other-value')
+        end
+
+        case facts[:osfamily]
+          when /Debian/
+
+            it 'writes config to /etc/default/marathon' do
+              should contain_file('/etc/default/marathon')
+                         .with_content(/^ulimit -n 9001$/)
+                         .with_content(/^JAVA_OPTS="-Xmx1024m -Dlogback.configurationFile=file:\/etc\/marathon\/logback.xml"$/)
+                         .with_content(/^JAVA_HOME="\/opt\/some\/java\/home"$/)
+                         .with_content(/^export foo="bar"$/)
+                         .with_content(/^export f00="b4r"$/)
+            end
+
+            it 'does not configure systemd' do
+              should_not contain_file('/etc/sysconfig/marathon')
+            end
+
+          when /RedHat/
+
+            it 'does not configure sysvinit' do
+              should contain_file('/etc/default/marathon')
+                         .with_ensure('absent')
+            end
+
+            it 'configures systemd' do
+              should contain_file('/etc/sysconfig/marathon')
+                         .with_content(/^JAVA_OPTS=-Xmx1024m -Dlogback.configurationFile=file:\/etc\/marathon\/logback.xml$/)
```

Comment:
That's actually valid systemd unit file syntax. Adding `"` would result in having `"-Xmx..."` as value.